src/lib/autocomplete/autocomplete.spec.ts

```diff
@@ -174,8 +174,17 @@ describe('MatAutocomplete', () => {
             .toBe(false, `Expected clicking outside the panel to set its state to closed.`);
         expect(overlayContainerElement.textContent)
             .toEqual('', `Expected clicking outside the panel to close the panel.`);
+
+        // Make sure that it works twice.
+        dispatchFakeEvent(input, 'focusin');
+        fixture.detectChanges();
+        expect(fixture.componentInstance.trigger.panelOpen).toBe(true);
+        return fixture.whenStable();
```

Comment:
This is slightly cleaner, but isn't immediately obvious that the `click` will wait for the second `whenStable`. Instead of returning the promise here, can you move the `click` event up and nest it in a `whenStable` instead?

Reply:
Done.